model-optimizer/extensions/middle/SplitConcatPairToInterpolate.py

```diff
@@ -53,7 +53,16 @@ def get_concat_after_split(split: Node) -> Optional[Node]:
 
     dest = split.out_port(0).get_destinations()[0].node
     # The transformation is applicable, only if next node is Concat.
-    return dest if dest.soft_get('type') == 'Concat' else None
+    if dest.soft_get('type') != 'Concat':
+        return
+
+    # The transformation is applicable, only if Split is a unique producer for Concat.
+    dest_inputs = [p.get_connection().get_source().node for _, p in dest.in_ports().items() if not p.disconnected()]
+    names_of_concat_inputs = set([n.name for n in dest_inputs])
+    if len(names_of_concat_inputs) != 1:
```

Comment:
I think it is sufficient to check `len(dest_inputs)`. Or even this logic can be much simpler, for example, check a number or connected input ports.

Reply:
It is not sufficient to check `len(dest_inputs)`, because number of connected input ports can be greater than 1, but corresponding producer nodes are the same (maybe, different input ports are corresponded to different output ports).

Reply:
But in your case you have already checked that `Split` has just one output port so `Concat` can not be producers twice or more

Reply:
I think I have got it. My wrong.